### ISSUE TITLE
Additional bars 2.01 crashed at startup while 1.01 works totally fine

### ISSUE THREAD

**Issue by heyjudek** [gamma1772/AdditionalBarsFabric#5]
[latest.log](https://github.com/gamma1772/AdditionalBarsFabric/files/6436849/latest.log)
[2021-05-06-7.log.gz](https://github.com/gamma1772/AdditionalBarsFabric/files/6436850/2021-05-06-7.log.gz)

**Comment by gamma1772** [comment]
I'm guessing its an issue with Full Slabs mod, but not 100% sure. I'll do some more testing tomorrow.

**Comment by gamma1772** [comment]
Not sure about the second issue though

**Comment by heyjudek** [comment]
Yes, removing additional bars makes the game launch normally. Why would a mod that has nothing to do with slabs (Additional bars) be conflicting with Full Slabs, though?
I'll try removing Third person maps and see how it works. But personally, I am not really fond of the idea of going back to an older Java version, for performance reasons.
In the meantime, may I ask what change you have made from 1.01 to 2.01 that would make the game crash?

**Comment by gamma1772** [comment]
Horizontal bars are slab blocks, so that's why its possible that there's conflict between two mods.
Code is the same between 1.0.1 and 2.0.1, except the newer version has a built-in resource pack, with code addition on client side (Which just loads that resource pack in the 'Resource packs' menu in game).

Difference in performance between java 15 and java 16 is negligible, since they both have the same increase in performance compared to Java 11. I've suggested J11 because it's LTS and I use it in dev environment.

**Comment by heyjudek** [comment]
Well, removing Third Person maps fixed the issue. But why this mod created a conflict is beyond me.
Full slabs just enables dynamically placing slab blocks vertically instead of creating a separate vertical equivalent. From what I saw, aren't the horizontal versions of Additional bars' blocks behaving the same way iron bars (a vanilla block) but with a different texture? Then I guess the problem must have come from the horizontal bars probably?

**Comment by gamma1772** [comment]
I also have no idea really, but I'll keep experimenting with these mods, probably something weird happening in the background that results in a crash.
Horizontal bars are a 'child' class of vanilla SlabBlock. They share some properties of slab blocks, but have their own shape, size, culling etc, while 'normal' bars and vanilla iron bars are just PaneBlocks with different models and textures. 

But yeah, errors can happen with mods that alter vanilla slab blocks code, for regular bars much less likely

I'll leave this open just in case, will close when I discover what truly causes the crash.

**Comment by heyjudek** [comment]
Sure, I appreciate your efforts. Regarding the 2.01, I really like the fact that wooden bars are now using planks' textures, in the meantime, you seemed to have significantly improved how crossed wooden bars look, which seem to no longer have those a single pixel thing edges. However, for some reason, crossed iron and gold bars look much worse regarding their shape compared to the older version.
